pkg/pool-utils/contracts/BaseMinimalSwapInfoPool.sol

```diff
@@ -38,7 +38,13 @@ abstract contract BaseMinimalSwapInfoPool is IMinimalSwapInfoPool, BasePool {
 
         if (request.kind == IVault.SwapKind.GIVEN_IN) {
             // Fees are subtracted before scaling, to reduce the complexity of the rounding direction analysis.
-            request.amount = _subtractSwapFeeAmount(request.amount);
+            uint256 amountInMinusSwapFees = _subtractSwapFeeAmount(request.amount);
+
+            // Process the (upscaled!) swap fee.
+            uint256 swapFee = request.amount - amountInMinusSwapFees;
```

Comment:
This sounds a bit awkward, isn't better to have another helper in the base pool that works as follows:

```solidity
uint256 swapFeeAmount = _getSwapFeeAmounts(request.amount);
uint256 amountInMinusSwapFees = request.amount.sub(swapFeeAmount);
```

Reply:
Yes, but that would require e.g. also changing `GeneralBasePool`, and I wanted to minimize disruption, especially considering we're going to revert these changes soon.